### PR TITLE
importccl: allow mysql AUTO_INCREMENT without start val

### LINKED ISSUE #29300: importccl: IMPORT MYSQLDUMP fails when AUTO_INCREMENT isn't set in table options
Testing with [this MySQL dump file](https://raw.githubusercontent.com/jethrocarr/amberphplib/40bbad29f8104c79506a802b569c4eb2559e1ab4/sql/amberphplib_template.sql) I found on GitHub, CockroachDB gave me this error:

```
root@:26257/defaultdb> import mysqldump ('https://raw.githubusercontent.com/jethrocarr/amberphplib/40bbad29f8104c79506a802b569c4eb2559e1ab4/sql/amberphplib_template.sql');
pq: column "id" specifies AUTO_INCREMENT but table options did not include it
```

The dump file includes a few table definitions that have an AUTO_INCREMENT `id` column; some of them have an AUTO_INCREMENT=xxx option in the table options, but others do not, and those are presumably the source of this error:

```sql
DROP TABLE IF EXISTS `file_upload_data`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `file_upload_data` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `fileid` int(11) NOT NULL DEFAULT '0',
  `data` blob NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COMMENT='Table for use as database-backed file storage system';
/*!40101 SET character_set_client = @saved_cs_client */;
```

### PR DESCRIPTION
**Pull request by dt**
for empty tables, mysqldump does not emit the AUTO_INCREMENT=X comment on the CREATE TABLE.
Previously we used this comment to determine if we'd make a sequence to allow auto_increment
default expressions on individual columns in the table definition.
However for empty tables, this meant those columns would then error since we thought we were
not going to support the sequence.

Instead, look for any auto_increments first to determine if we need the sequence, then handle
the individual column defs.

Release note: none.

Fixes #29300.

### CODE REVIEW

File: pkg/ccl/importccl/testdata/mysqldump/third.cockroach-schema.sql [hunk 1]
```diff
@@ -1,5 +1,7 @@
+CREATE SEQUENCe third_auto_inc;
```
Comment (mjibson) [inline pkg/ccl/importccl/testdata/mysqldump/third.cockroach-schema.sql]: SEQUENCe

File: pkg/ccl/importccl/read_import_mysql.go [hunk 2]
```diff
@@ -345,9 +362,7 @@ func mysqlTableToCockroach(
 			return nil, nil, err
 		}
 		if raw.Type.Autoincrement {
-			if seqName == "" {
```
Comment (mjibson) [inline pkg/ccl/importccl/read_import_mysql.go]: Is it possible for autoincrement to appear twice in a table?

**Review by madelynnblue — APPROVED** [review]

**Review by rolandcrosby — APPROVED** [review]
Thanks for the quick fix! This appears to be one of the more common incompatibilities we have with the MySQL dump files I've seen in the wild, so this should help a lot.

File: pkg/ccl/importccl/read_import_mysql.go (hunk 2 above)
Comment (dt) [reply pkg/ccl/importccl/read_import_mysql.go]: no

File: pkg/ccl/importccl/testdata/mysqldump/third.cockroach-schema.sql (hunk 1 above)
Comment (dt) [reply pkg/ccl/importccl/testdata/mysqldump/third.cockroach-schema.sql]: Done.

**Review by madelynnblue — APPROVED** [review]

**Pull request merged**